src/analyses/useAfterFree.ml

```diff
@@ -0,0 +1,188 @@
+(** An analysis for the detection of use-after-free vulnerabilities. *)
+
+open GoblintCil
+open Analyses
+
+module Spec : Analyses.MCPSpec =
+struct
+  include Analyses.DefaultSpec
+
+  let name () = "useafterfree"
+  (* module D = Lattice.Unit *)
+  module D = SetDomain.Make(ValueDomain.Blobs)
+  module C = Lattice.Unit
+
+  (** TODO: What about context-sensitivity? *)
+  let context _ _ = ()
+
+
+  (* HELPER FUNCTIONS *)
+
+  let check_exp (exp:exp) ctx =
```

Comment:
Here, we would also need to check intermediate expressions. E.g.  `a[*p] = 5` constitutes a UAF if `p` points to dynamically allocated memory, that has been freed.